packages/SwingSet/src/kernel/vatManager/vat-warehouse.js

```diff
@@ -1,26 +1,82 @@
 // @ts-check
-import { assert } from '@agoric/assert';
+import { assert, details as X, quote as q } from '@agoric/assert';
 import { makeVatTranslators } from '../vatTranslator.js';
 
+const { freeze } = Object;
+
+/** @param { number } max */
+export const makeLRU = max => {
+  /** @type { string[] } */
+  const items = [];
+
+  return freeze({
+    /** @param { string } item */
+    add: item => {
+      const pos = items.indexOf(item);
+      // already most recently used
+      if (pos + 1 === max) {
+        return null;
+      }
+      // remove from former position
+      if (pos >= 0) {
+        items.splice(pos, 1);
+      }
+      items.push(item);
+      // not yet full
+      if (items.length <= max) {
+        return null;
+      }
+      const [lru] = items.splice(0, 1);
```

Comment:
nit: I get confused between LRU the data structure (with a name like `makeLRU` I expect clients to store it in `const lru = makeLRU()`) and `lru` the thing being removed from the data structure because it was the least-recently-used. Since the item being returned has already been removed, maybe name it `dropped` or `removed`?